pkg/operatorsource/configuring.go

```diff
@@ -95,20 +132,86 @@ func (r *configuringReconciler) Reconcile(ctx context.Context, in *marketplace.O
 	}
 
 	// If we are here, the given CatalogSourceConfig object already exists.
-	cscNamespacedName := types.NamespacedName{Name: in.Name, Namespace: in.Namespace}
-	cscExisting := marketplace.CatalogSourceConfig{}
-	err = r.client.Get(ctx, cscNamespacedName, &cscExisting)
+	err = r.updateExistingCatalogSourceConfig(ctx, in, packages)
 	if err != nil {
 		r.logger.Errorf("Unexpected error while getting CatalogSourceConfig: %s", err.Error())
 		nextPhase = phase.GetNextWithMessage(phase.Configuring, err.Error())
-
 		return
 	}
 
+	r.logger.Info("CatalogSourceConfig object has been updated successfully")
+
+	nextPhase = phase.GetNext(phase.Succeeded)
+	return
+}
+
+// downloadManifests abstracts away the download of package metadata from the operatorsource endpoint
+// it returns the list of packages to be written to the operatorsource status and an error
+func (r *configuringReconciler) downloadManifests(spec *marketplace.OperatorSourceSpec, namespace string) ([]*datastore.RegistryMetadata, error) {
+
+	manifests := make([]*datastore.RegistryMetadata, 0)
+
+	options, err := SetupAppRegistryOptions(r.client, spec, namespace)
+	if err != nil {
+		return manifests, err
+	}
+
+	registry, err := r.factory.New(options)
+	if err != nil {
+		return manifests, err
+	}
+
+	manifests, err = registry.ListPackages(spec.RegistryNamespace)
+	if err != nil {
+		return manifests, err
+	}
+
+	return manifests, nil
+}
+
+// checkDatastoreExistingPackages checks to see if there are any packages before we write to the datastore
+// If there are not, we are assuming this is a new operator source. We should force all catalogsourceconfigs
```

Comment:
Please us `OperatorSource` and `CatalogSourceConfig`. Comment applies to all places.